File: FEBRUARY/04-02-2026/3640.java
class Solution {
    public long maxSumTrionic(int[] nums) {
        int n= nums.length;
        long presum [] = new long [n+1];

        for(int i=1; i<=n;i++){
            presum[i]= presum[i-1] + nums[i-1];
        }
        long maxsum =Long.MIN_VALUE;
        int i=1;

        while(i< n-2){
            int left = i-1;

            while( i< n && nums[i] > nums[i-1])
            i++;
            if(i == left +1){
                i++;
                continue;
            }
            int pIndex= i-1;

            while(i< n && nums[i] < nums[i-1])
            i++;
            if(i== pIndex +1){
                i++;
                continue;
            }
            int qIndex= i-1;

            while( i< n && nums[i] > nums[i-1])
            i++;
            if(i==qIndex +1){
                i++;
                continue;
            }


            int right = i-1;

            long maxprefixsum= Long.MIN_VALUE;
            long  minprefixsum= Long.MAX_VALUE;

            for(int j= left; j<pIndex;j++){
                minprefixsum= Math.min(minprefixsum, presum[j]);
            }
            for(int j=qIndex+1; j<=right;j++){
                maxprefixsum= Math.max(maxprefixsum, presum[j+1]);
            }
            maxsum= Math.max(maxsum, maxprefixsum - minprefixsum);
            i= qIndex +1;
        }
        return maxsum;

    }
}

File: 2026POTD/FEBRUARY/04-02-2026/3640.cpp

class Solution {
public:
    long long maxSumTrionic(vector<int>& nums) {
        int n = nums.size();
        int state=0;
        long long count[3]={0};
        long long ans = -1e14;
        count[0]=nums[0];
        for (int i=1; i<n; i++)
        {
            if(nums[i]==nums[i-1])
            {
                count[0] = nums[i];
                state = 0;
                continue;
            } 
            switch(state)
            {
                case 0:
                if(nums[i]>nums[i-1])
                {
                    count[0] += nums[i];
                     state=1;
                }
                else 
                {
                    count[0]=nums[i];
                }
                break;
                case 1:
                if(nums[i]<nums[i-1])
                {
                    count[1] = count[0] + nums[i]; // nums[i]+nums[i-1];
                    state=2;
                }
                else
                {
                    count[0]+=nums[i];
                    if (count[0]<nums[i]+nums[i-1])
                      count[0] = nums[i]+nums[i-1];
                }
                break;
                case 2:
                if(nums[i]>nums[i-1])
                {
                    count[2] = count[1] + nums[i]; // nums[i]+nums[i-1];
                    count[0] = nums[i] + nums[i-1];
                    ans = max(ans,count[2]);
                    state=3;
                }
                else
                {
                    count[1]+=nums[i];
                }
                break;

                case 3:
                if(nums[i]<nums[i-1])
                {
                    count[1] = count[0] + nums[i];
                    state=2;
                }
                else
                {
                    count[2]+=nums[i];
                    count[0]+=nums[i];
                    if (count[0]<nums[i]+nums[i-1])
                      count[0] = nums[i]+nums[i-1];
                    ans = max(ans,count[2]);
                }
                break;
            }
        }
        return ans;
    }
};